src/nom.rs

```diff
@@ -602,9 +602,10 @@ pub fn float(input: &[u8]) -> IResult<&[u8],f32> {
     recognize!(
       tuple!(
         opt!(alt!(tag!("+") | tag!("-"))),
-        alt!(
-          delimited!(digit, tag!("."), opt!(digit))
-          | delimited!(opt!(digit), tag!("."), digit)
+        alt_complete!(
```

Comment:
Nom already does that with IResult::Incomplete.  But in this case, the buffer ends and the parser is at a state where it *could* be finished: think of reading "123.45" at the end of the buffer, but the next characters (which are unavailable yet) are "67".

digit() already has a check for input_length == 0 when it starts right at the end of the buffer.  Maybe if it runs out of buffer in the for loop it should also return Incomplete?

(What would happen after that, both for the "here's some more data" case and for the "nope, that's the only buffer we'll have" case?)